pkg/solana/txm/txm.go

```diff
@@ -377,135 +382,220 @@ func (txm *Txm) confirm() {
 		case <-ctx.Done():
 			return
 		case <-tick:
-			// get list of tx signatures to confirm
-			sigs := txm.txs.ListAll()
-
-			// exit switch if not txs to confirm
-			if len(sigs) == 0 {
+			// If no signatures to confirm and rebroadcast, we can break loop as there's nothing to process.
+			if txm.InflightTxs() == 0 {
 				break
 			}
 
-			// get client
 			client, err := txm.client.Get()
 			if err != nil {
-				txm.lggr.Errorw("failed to get client in soltxm.confirm", "error", err)
-				break // exit switch
+				txm.lggr.Errorw("failed to get client in txm.confirm", "error", err)
+				break
 			}
+			txm.processConfirmations(ctx, client)
 
-			// batch sigs no more than MaxSigsToConfirm each
-			sigsBatch, err := utils.BatchSplit(sigs, MaxSigsToConfirm)
-			if err != nil { // this should never happen
-				txm.lggr.Fatalw("failed to batch signatures", "error", err)
-				break // exit switch
+			// In case all txes where confirmed and there's nothing to rebroadcast.
+			// This check saves making 2 RPC calls (slot height + blockhash) when there's nothing to process.
+			// Passing MaxUint64 as currHeight to ListAllExpiredBroadcastedTxs will return all broadcasted txs.
+			if len(txm.txs.ListAllExpiredBroadcastedTxs(math.MaxUint64)) == 0 {
```

Comment:
If we move the blockhash logic to `sendWithRetry`, this would only help avoid the 1 RPC call for slot height. I think it's good we're conscious about RPC calls but think it's probably extremely rare we don't have something in broadcast state at any given time. And if we have quite a few, this might be doing more work than it's saving since it'd parse every broadcasted tx and return it. I think we can nix this check. If you think it's helping more than I'm thinking, we should at least move it into `rebroadcastExpiredTxs` so we're not doing the extra validation if the feature is turned off.

Reply:
Yes, maybe was trying to overengineer this. Let's keep it simple right now and remove it